### ISSUE TITLE
ios safari don't support it

### ISSUE THREAD

**Issue by duochizhacai** [fregante/content-scripts-register-polyfill#70]
I am working on safari extension with ios 16.3.1,while registerContentScript came to an error
here comes the code
![image](https://user-images.githubusercontent.com/58379214/230441704-39c6f370-dbbe-4252-a40d-141904646783.png)
there are an error in the console
![image](https://user-images.githubusercontent.com/58379214/230441133-9c971152-4bd6-49e5-9eb7-c56e1a7ea199.png)
How should I pass parameters when using registerContentScript？

**Comment by duochizhacai** [comment]
it still happen when i delete the allFrames parameter!!!😭

**Comment by fregante** [comment]
Note the error. It's related to the scripting API, not this API

**Comment by fregante** [comment]
One thing to note is that iOS 16.4 supports the new scripting.registerContentScript API so it has already been fixed there I think. 

I'll take a look at this when I can

**Comment by fregante** [comment]
Are you sure that it's coming from this package because the allFrame is discarded here, check the code:

https://github.com/fregante/content-scripts-register-polyfill/blob/main/ponyfill.ts

Additionally, as per the warning in the readme:

> **Warning** `contentScripts.register` by design is not compatible with Event Pages (MV2) and Service Workers (MV3).

Safari iOS cannot support this API because it only has Event pages (persistent: false). The API, as designed by Mozilla, simply isn't compatible with them. 

In short: what you're trying to achieve is only possible in iOS 16.4 and you don't need to use this API/polyfill in any browser (scripting.registerContentScript is supported everywhere)
